### PR TITLE
fix: (platform) Link documentation changes and Extending from Base class.

### PR DESCRIPTION
**Pull request by DeepakSap14**
#### Please provide a link to the associated issue.
mentioned in https://github.com/SAP/fundamental-ngx/issues/2542

#### Please provide a brief summary of this pull request.
1). It has documentation details updated.
2). added type attribute for setting native type attribute on <a> tag.
3). Added examples for added attributes.

BREAKING CHANGES:
Renamed properties:
Before :type After: linktype

#### Please check whether the PR fulfills the following requirements

- [x] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/master/CONTRIBUTING.md
- [x] tests for the changes that have been done
- [x] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist

Documentation checklist:
- [x] update `README.md`
- [x] Documentation Examples
- [x] Stackblitz works for all examples

### CODE REVIEW

File: apps/docs/src/app/platform/component-docs/platform-link/platform-link-header/platform-link-header.component.html [hunk 1]
```diff
@@ -1,8 +1,12 @@
-<header>Link</header>
+<header>Link Component</header>
 <description>
-    The Link Component is used to enhance standard HTML link.
-    <br /><br />
-    Link can be clicked by keyboard enter key and Mouse click.
+    Link in Platform is functionally similar to the Link in
+    <a href="https://sap.github.io/fundamental-ngx/#/core/link">fundamental-ngx/core</a>. In Platform, Link is wrapped
```
Comment (kavya-b) [inline apps/docs/src/app/platform/component-docs/platform-link/platform-link-header/platform-link-header.component.html:4]: this is fine, just fyi, you can even use `<a [routerLink]="'/core/link'">` instead

Comment (DeepakSap14) [reply apps/docs/src/app/platform/component-docs/platform-link/platform-link-header/platform-link-header.component.html]: Done.

**Comment by DeepakSap14** [comment]
> The PR looks good to me. It doesn't cover all the points listed in #2542

Now added examples to cover Issue #2542.
extended from base class.
renamed "type" property to "linktype", as it was overlapping with native "type" for anchor tag.

File: libs/platform/src/lib/components/link/link.component.ts [hunk 2]
```diff
@@ -10,51 +11,59 @@ import {
...
+     * sets target to Native anchor.
+     */
     @Input()
     target?: NavigationTarget;
 
-    /** type of link, options standard or Emphasized, Default=standard */
+    /** linktype of link, options standard or Emphasized, Default=standard */
     @Input()
-    type: LinkType = 'standard';
+    linktype?: LinkType = 'standard';
```
Comment (KevinOkamoto) [inline libs/platform/src/lib/components/link/link.component.ts:55]: Use camel-case, change `linktype` to `linkType`.

File: libs/platform/src/lib/components/link/link.component.ts [hunk 3]
```diff
@@ -10,51 +11,59 @@ import {
...
 
+    /**
+     * type of link. possible values text|application|audio|font|example|image|message|model|multipart|video.
+     * sets type to Native anchor.
+     */
     @Input()
-    get disabled(): boolean {
```
Comment (stefanoScalzo) [inline libs/platform/src/lib/components/link/link.component.ts:49]: why are you removing these methods?

File: libs/platform/src/lib/components/link/link.component.ts (hunk 2 above)
Comment (DeepakSap14) [reply libs/platform/src/lib/components/link/link.component.ts]: Done

File: libs/platform/src/lib/components/link/link.component.ts (hunk 3 above)
Comment (DeepakSap14) [reply libs/platform/src/lib/components/link/link.component.ts:49]: Link Component is extending BaseComponent. BaseComponent has disabled property defined. So removed duplication from Link Component.

Comment (fkolar) [reply libs/platform/src/lib/components/link/link.component.ts:49]: Disabled state is almost on every single component

**Comment by DeepakSap14** [comment]
> Just camel-case `linkType`, otherwise it looks good.

changes are done and pushed.

**Pull request merged**